pkg/util/openstack/client.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+	"gopkg.in/gcfg.v1"
+)
+
+type KeystoneAuthOpts struct {
```

Comment:
It was pulled out from existing code in Kubernetes that authenticated with OpenStack to do stuff like volume and load balancer management and made generic enough that it can be shared between the CloudProviders use case and the Keystone Authn/z plugin. The more specific code was in pkg/cloudprovider/providers/openstack/openstack.go